phpunit/functional/SLMTest.php

```diff
@@ -2226,4 +2532,34 @@ public function testCloneOLA()
             $ola_compare
         );
     }
+
+    /**
+     * Set $_SESSION['glpi_currenttime'] with current day + $time param and return the related DateTime
+     *
+     * @param string $time expected format is H:i:s
+     */
+    private function setCurrentTime(string $time): \DateTime
+    {
+        // assert format is H:i:s
+        assert(1 === preg_match('/^([01]?[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])$/', $time));
+
+        // set session time
+        $_SESSION['glpi_currenttime'] = date('Y-m-d ' . $time);
+
+        // create and return DateTime
+        $dt = new \DateTime();
+        $dt->setTime(...explode(':', $time));
+
+        return $dt;
+    }
```

Comment:
You can move this in the `GLPITestCase` class. You should also pass a date in the parameters, to be able tio reuse the default calendar and make your tests work during holidays.

Reply:
Also `date('Y-m-d ' . $time)` could lead to some flakiness as the day is not guaranteed to be the same if the tests are run close to midnight (even if it is quite unlikely to ever happen).